helmfile/charts/log-manager/files/compaction.sh

```diff
@@ -85,14 +85,16 @@ merge_chunks() {
 
       echo "s3://$S3_BUCKET/$S3_PREFIX/$DAY/${FILE/$LM_TMP\//}" >> "$TMPFILE.idx"
 
-      zstd -c -d --rm "$FILE" >> "$TMPFILE.log"
+      zstd -c -d "$FILE" >> "$TMPFILE.log"
+      rm "$FILE"
     done
 
     echo "----- sorting chunk"
     sort --temporary-directory="$SORT_TMP" -u -S 100M -o "$TMPFILE.log" "$TMPFILE.log"
 
     echo "----- compressing chunk"
```

Comment:
```suggestion
      zstd -c -d "$FILE"
      rm "$FILE" >&2
    done | sort --compress-program=zstd --temporary-directory="$SORT_TMP" -u -S 100M | zstd -o "$TMPFILE.zst"

    echo "----- compressing chunk"
```

What if we do everything on the fly?

Reply:
It will require a lot of cpu time but the size gain would be huge. We would additionally need to increase the timeout for the job slightly to cope with large chunks.

Reply:
Testing with what yielded a ~50GiB uncompressed chunk that would have used min 100GiB max 150GiB scratch storage used only ~12GiB. So big gains.

Reply:
💪🏻

Reply:
@aarnq What do you think of f9945e64257961dc5a2c79c694f6e651a7334756